rlp/sedes/dynamic.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, List
+
+from rlp.exceptions import (
+    ObjectDeserializationError,
+    ObjectSerializationError,
+)
+
+
+class DynamicSerializable(ABC):
+    @property
+    @abstractmethod
+    def sedes_options(self) -> List:
+        ...
+
+    @classmethod
+    def deserialize(cls, obj: Any) -> bytes:
+        for sedes in cls.sedes_options:
+            try:
+                return sedes.deserialize(obj)
+            except ObjectDeserializationError:
+                pass
```

Comment:
What happens if all of the sedes options result in `ObjectDeserializationError`.  Won't that result in this returning `None`?

Reply:
Yeah, sorry I "marked for review" a bit early. That's one of the TODOs listed above.